pkg/scheduler/queue/queue_test.go

```diff
@@ -622,6 +622,48 @@ func TestRequestQueue_GetNextRequestForQuerier_ShouldReturnImmediatelyIfQuerierI
 	require.EqualError(t, err, "querier has informed the scheduler it is shutting down")
 }
 
+func TestRequestQueue_GetNextRequestForQuerier_ShouldReturnErrorIfCtxCanceled(t *testing.T) {
+	// This tests a rare (~5 in 100K test runs) race where a canceled querier
```

Comment:
I am not sure I understand this - the test always passes now that the fix is in, but it almost always passes without the fix too?

Reply:
Yeah, the old "add a failing test and fix it" dance. It's just a test that didn't fail 100% of the time.

Reply:
Perhaps no comment would be better, and future archeologists can follow the PR history.

Reply:
in wondering whether we should a have test at all that passes 99% of the time even when the fix isn’t there.
but I do not feel strongly either way

Reply:
It just becomes coverage for a codepath that wasn't quite tested before, I don't think. I've just expanded one of your existing tests to also exercise that code.